### PR TITLE
Create AWS Security Hub Rules

### PR DESCRIPTION
**Pull request by IsExec**
|Related issue|
|---|
|#22505|
 
- [x] Added base ruleset for AWS security hub:
    - [x] Security hub controls base rules
    - [x] Base rules for AWS service integrations (GuardDuty, Inspector and Health)
    - [x] Rules for critical security control checks
- [x] Create .ini check file

- Decoder/Rule tests
  - [x] Added unit testing files ".ini"
  - [x] runtests.py executed without errors

### CODE REVIEW

**Comment by IsExec** [comment]
```
root@ubuntu2204:/tmp/wazuh/ruleset/testing# python2 runtests.py --testfile aws_security_hub.ini
- [ File = ./tests/aws_security_hub.ini ] ---------
.........................

```

|Component |  Tested  |  Total   | Coverage |
| -------- | -------- | -------- | -------- |
|  Rules   |    25    |   4481   |  0.56%   |
| Decoders |    1     |   170    |  0.59%   |


|          File           |  Passed  |  Failed  |  Status  |
|        --------         | -------- | -------- | -------- |
|./tests/aws_security_hub.ini|    25    |    0     |   ✅    |

**Comment by Johnng007** [comment]
### Update 10/05/2024

- Group name should end with a comma. i.e `<group name="amazon,securityhub,">`

- The compliance standard group shoould also end with a comma i.e `<group>pci_dss_8.2.1,nist_800_53_IA.5,nist_800_53_SA.3</group>`

File: ruleset/testing/tests/aws_security_hub.ini
Comment (Selutario) [inline ruleset/testing/tests/aws_security_hub.ini:1]: I see some actual information included in this file. Would this work if the account IDs were replaced with some value like 123456789123, and in the same way for bucket names and other resources?

**Review by Selutario — COMMENTED** [review]
This looks pretty good to me, just a minor request.

File: ruleset/testing/tests/aws_security_hub.ini
Comment (IsExec) [reply ruleset/testing/tests/aws_security_hub.ini:1]: Actual information have been replaced with dummy - [84cc9da](https://github.com/wazuh/wazuh/pull/23234/commits/84cc9daf8360d5df30d285210cdd62242cbf1b7e)

**Review by Selutario — APPROVED** [review]

**Review by Johnng007 — COMMENTED** [review]
LGTM!

**Review by Johnng007 — APPROVED** [review]

File: ruleset/rules/0998-aws-security-hub-rules.xml [hunk 1]
```diff
@@ -0,0 +1,491 @@
+<!--
+  -  AWS Security Hub default ruleset
+  -  Created by Wazuh, Inc.
+  -  Copyright (C) 2024, Wazuh Inc.
+  -  This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2.
+  ID: 99800 - 99860
+-->
+
+<group name="amazon,securityhub,">
```
Comment (ooniagbi) [inline ruleset/rules/0998-aws-security-hub-rules.xml:9]: Change group name to `aws_security_hub`. Use only one group name. Similar to the screenshot:
image

File: ruleset/rules/0998-aws-security-hub-rules.xml [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+      <id>T1530</id>
+    </mitre>
+  </rule>
+
+  <rule id="99845" level="12">
+    <if_sid>99821</if_sid>
+    <field name="aws.finding.Compliance.SecurityControlId">^IAM.4$</field>
+    <field name="aws.finding.Compliance.Status">FAILED</field>
+    <field name="aws.finding.RecordState">ACTIVE</field>
+    <description>AWS Security Hub - IAM root user access key exist.</description>
```
Comment (ooniagbi) [inline ruleset/rules/0998-aws-security-hub-rules.xml:339]: exists

File: ruleset/rules/0998-aws-security-hub-rules.xml [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+      <id>T1556.006</id>
+    </mitre>
+  </rule>
+
+  <rule id="99848" level="12">
+    <if_sid>99822</if_sid>
+    <field name="aws.finding.Compliance.SecurityControlId">^KMS.3$</field>
+    <field name="aws.finding.Compliance.Status">FAILED</field>
+    <field name="aws.finding.RecordState">ACTIVE</field>
+    <description>AWS Security Hub - AWS KMS keys is scheduled for deletion.</description>
```
Comment (ooniagbi) [inline ruleset/rules/0998-aws-security-hub-rules.xml:372]: Remove the "is".

File: ruleset/rules/0998-aws-security-hub-rules.xml [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+    <description>AWS Security Hub - Amazon S3 general purpose bucket permits public write access.</description>
+    <group>pci_dss_1.2.1,pci_dss_1.3.1,pci_dss_1.3.2,pci_dss_1.3.4,pci_dss_1.3.6,pci_dss_7.2.1,nist_800_53_AC.3,nist_800_53_AC.4,nist_800_53_AC.6,nist_800_53_AC.21,nist_800_53_SC.7,</group>
+  </rule>
+
+  <rule id="99857" level="12">
+    <if_sid>99829</if_sid>
+    <field name="aws.finding.Compliance.SecurityControlId">^S3.19$</field>
+    <field name="aws.finding.Compliance.Status">FAILED</field>
+    <field name="aws.finding.RecordState">ACTIVE</field>
+    <description>AWS Security Hub - Amazon S3 access point block public access settings aren't enabled.</description>
```
Comment (ooniagbi) [inline ruleset/rules/0998-aws-security-hub-rules.xml:456]: Change to "are not".

**Review by ooniagbi — CHANGES_REQUESTED** [review]
Please make the requested changes to the affected rules in the ini file also.

File: ruleset/rules/0998-aws-security-hub-rules.xml (hunk 1 above)
Comment (IsExec) [reply ruleset/rules/0998-aws-security-hub-rules.xml]: This has been resolved https://github.com/wazuh/wazuh/pull/23234/commits/1d4bfae8c2701e7cc0cda7b3f8df0965bdb225f8.

File: ruleset/rules/0998-aws-security-hub-rules.xml (hunk 4 above)
Comment (IsExec) [reply ruleset/rules/0998-aws-security-hub-rules.xml]: This has been resolved https://github.com/wazuh/wazuh/pull/23234/commits/1d4bfae8c2701e7cc0cda7b3f8df0965bdb225f8.

File: ruleset/rules/0998-aws-security-hub-rules.xml (hunk 2 above)
Comment (IsExec) [reply ruleset/rules/0998-aws-security-hub-rules.xml]: This has been resolved https://github.com/wazuh/wazuh/pull/23234/commits/1d4bfae8c2701e7cc0cda7b3f8df0965bdb225f8.

File: ruleset/rules/0998-aws-security-hub-rules.xml (hunk 3 above)
Comment (IsExec) [reply ruleset/rules/0998-aws-security-hub-rules.xml]: This has been resolved https://github.com/wazuh/wazuh/pull/23234/commits/1d4bfae8c2701e7cc0cda7b3f8df0965bdb225f8.

**Review by ooniagbi — APPROVED** [review]
LGTM!

**Pull request merged**